### PR TITLE
Implement get_quota and initial implementation of validate_allocations

### PR DESCRIPTION
**Pull request by knikolla**
- The ResourceAllocator base class now defines get_quota and the OpenStack
resource allocator implements the method.

- Currently limited to OpenStack resources, the `validate_allocations` command
checks that the output of get_quota on an allocation, matches the value of the quota
attributes.

### CODE REVIEW

**Comment by joachimweyl** [comment]
@knikolla is it clear what needs to be fixed/updated for test-py39-functional / build (pull_request)? Are you ready for Reviewers to be added to this PR?

**Comment by knikolla** [comment]
@joachimweyl I know why the test is failing, It's on my plate to fix it in the next 2 hours.

**Comment by joachimweyl** [comment]
@knikolla now that this passes the checks, is this ready for reviewers?

File: src/coldfront_plugin_openstack/management/commands/validate_allocations.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                        msg = f'Attribute "{attr}" expected on allocation {allocation_str} but not set.'
+                        logger.warning(msg)
+                    else:
+                        key = openstack.QUOTA_KEY_MAPPING_ALL_KEYS.get(attr, None)
+                        if not key:
+                            # Note(knikolla): Some attributes are only maintained
+                            # for bookkeeping purposes and do not have a
+                            # corresponding quota set on the service.
+                            continue
+                        elif not (value := quota.get(key, None)) == expected_value:
```
[the remark this answers is not in the document]
Comment (knikolla) [reply src/coldfront_plugin_openstack/management/commands/validate_allocations.py:60]: haha, any suggestions on how to better write this?

[the remark this answers is not in the document]
Comment (larsks) [reply src/coldfront_plugin_openstack/management/commands/validate_allocations.py:60]: No, this is fine, just struck me!

File: src/coldfront_plugin_openstack/openstack.py [hunk 2]
```diff
@@ -16,6 +16,8 @@
 
 logger = logging.getLogger(__name__)
 
+# 1 GB = 1 000 000 000 B = 10^9 B
+GB_IN_BYTES = 1000000000
```
Comment (naved001) [inline src/coldfront_plugin_openstack/openstack.py:20]: probably just a preference thing but I would write this as `10**9` to not accidentally add more/less zeros.

**Review by naved001 — APPROVED** [review]

**Comment by joachimweyl** [comment]
@knikolla do we need @jtriley to take a look or is this ready to be merged?

File: src/coldfront_plugin_openstack/openstack.py [hunk 3]
```diff
@@ -108,38 +140,49 @@ def set_quota(self, project_id):
...
+                    self.object(project_id).post_account(headers=payload)
                 except ksa_exceptions.NotFound:
                     logger.debug('No swift available, skipping its quota.')
 
+    def get_quota(self, project_id):
+        quotas = dict()
+
+        compute_quota = self.compute.quotas.get(project_id)
+        for k in QUOTA_KEY_MAPPING['compute']['keys'].values():
+            quotas[k] = compute_quota.__getattr__(k)
```
Comment (jtriley) [inline src/coldfront_plugin_openstack/openstack.py:165]: Can we not use `get()` vs `__getattr__()` here for `compute_quota` and `volume_quota` as you're doing for `network_quota`?

**Review by jtriley — CHANGES_REQUESTED** [review]
LGTM just need to address the two minor comments.

File: src/coldfront_plugin_openstack/openstack.py (hunk 2 above)
Comment (knikolla) [reply src/coldfront_plugin_openstack/openstack.py:20]: Makes sense. I considered doing that.

**Comment by knikolla** [comment]
@jtriley can you update your review to approved and merge?

**Review by jtriley — APPROVED** [review]

**Pull request merged**